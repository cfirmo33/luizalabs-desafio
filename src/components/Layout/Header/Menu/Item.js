import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import If from '../../../Layout/Helper/If'
import { Link, withRouter } from 'react-router-dom'

const isActive = (match, to) => match.path === to

const Item = ({
  title,
  number,
  to,
  match,
}) => (
  <li>
    <Link to={to} className={classnames({ active: isActive(match, to) })}>
      {title}
      <If test={(number !== undefined && number > 0)}>
        <small>{number}</small>
      </If>
    </Link>
  </li>
)

Item.propTypes = {
  title: PropTypes.string.isRequired,
  number: PropTypes.number,
  to: PropTypes.string.isRequired,
  match: PropTypes.objectOf(Object).isRequired,
}

Item.defaultProps = {
  number: 0,
}

export default withRouter(Item)
