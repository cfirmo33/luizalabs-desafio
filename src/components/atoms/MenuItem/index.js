import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'
import If from '../If'

const MenuItem = ({ title, active, number }) => (
  <li>
    <a href=":;" className={classnames({ active })}>
      {title}
      <If test={number}>
        <small>{number}</small>
      </If>
    </a>
  </li>
)

MenuItem.propTypes = {
  title: PropTypes.string.isRequired,
  active: PropTypes.bool,
  number: PropTypes.number,
}

MenuItem.defaultProps = {
  active: false,
}

export default MenuItem
