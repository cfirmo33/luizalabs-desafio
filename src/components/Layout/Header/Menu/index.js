import React from 'react'
import PropTypes from 'prop-types'
import MenuItem from './Item'

const Menu = ({ children }) => (
  <nav>
    <ul>
      {children}
    </ul>
  </nav>
)

Menu.propTypes = {
  children: PropTypes.arrayOf(MenuItem).isRequired,
}

export default Menu
