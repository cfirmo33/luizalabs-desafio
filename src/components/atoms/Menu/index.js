import React from 'react'
import PropTypes from 'prop-types'
import MenuItem from '../MenuItem'

const Menu = ({ children }) => (
  <nav>
    <ul>
      {children}
    </ul>
  </nav>
)

Menu.propTypes = {
  children: PropTypes.arrayOf(MenuItem).isRequired,
}

export default Menu
