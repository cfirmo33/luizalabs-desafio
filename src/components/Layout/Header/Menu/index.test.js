import React from 'react'
import { shallow } from 'enzyme'
import MenuItem from './Item'
import Menu from './index'


describe('Menu', () => {
  it('Deve renderizar corretamente', () => {
    const match = {
      path: '/',
    }
    const wrapper = shallow(
      <Menu>
        <MenuItem title="Home" to="/" match={match} />
        <MenuItem title="Total" number={10} to="/" match={match} />
      </Menu>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
