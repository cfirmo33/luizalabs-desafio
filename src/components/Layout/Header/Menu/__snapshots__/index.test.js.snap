// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Deve renderizar corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu>
    <withRouter(Item)
        match={
            Object {
                "path": "/",
              }
        }
        title="Home"
        to="/"
    />
    <withRouter(Item)
        match={
            Object {
                "path": "/",
              }
        }
        number={10}
        title="Total"
        to="/"
    />
</Menu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <withRouter(Item)
                match={
                        Object {
                                "path": "/",
                              }
                }
                title="Home"
                to="/"
        />
        <withRouter(Item)
                match={
                        Object {
                                "path": "/",
                              }
                }
                number={10}
                title="Total"
                to="/"
        />
</ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <withRouter(Item)
            match={
                        Object {
                                    "path": "/",
                                  }
            }
            title="Home"
            to="/"
/>,
          <withRouter(Item)
            match={
                        Object {
                                    "path": "/",
                                  }
            }
            number={10}
            title="Total"
            to="/"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "match": Object {
              "path": "/",
            },
            "title": "Home",
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "match": Object {
              "path": "/",
            },
            "number": 10,
            "title": "Total",
            "to": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <withRouter(Item)
                    match={
                              Object {
                                        "path": "/",
                                      }
                    }
                    title="Home"
                    to="/"
          />
          <withRouter(Item)
                    match={
                              Object {
                                        "path": "/",
                                      }
                    }
                    number={10}
                    title="Total"
                    to="/"
          />
</ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <withRouter(Item)
              match={
                            Object {
                                          "path": "/",
                                        }
              }
              title="Home"
              to="/"
/>,
            <withRouter(Item)
              match={
                            Object {
                                          "path": "/",
                                        }
              }
              number={10}
              title="Total"
              to="/"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "match": Object {
                "path": "/",
              },
              "title": "Home",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "match": Object {
                "path": "/",
              },
              "number": 10,
              "title": "Total",
              "to": "/",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
